packages/tc-api-rest-handlers/absorb.js

```diff
@@ -220,12 +242,42 @@ const absorbHandler = ({ documentStore } = {}) => async input => {
 		// Merged result always exists at last index
 		result = results.pop();
 
-		broadcast('UPDATE', result, {
-			relationships: {
-				removed: removedRelationships,
-			},
+		const addedRelationships = getAddedRelationships({
+			type: nodeType,
+			initialContent: mainRecord,
+			newContent: result.toJson({ type: nodeType, excludeMeta: true }),
 		});
-		broadcast('DELETE', absorbedNode);
+
+		broadcast([
+			{
+				action: 'DELETE',
+				type: nodeType,
+				code: absorbedCode,
+			},
+			{
+				action: 'UPDATE',
+				code: absorbedCode,
+				type: nodeType,
+				removedRelationships: getAlteredPeers({
+					properties,
+					absorbedRecord,
+				}),
+				neo4jResult: results[0],
+			},
+			{
+				action: 'UPDATE',
+				code,
+				type: nodeType,
+				addedRelationships,
+				removedRelationships,
+				updatedProperties: [
+					...Object.keys(writeProperties),
+					...Object.keys(addedRelationships),
+					...(updatedDocumentProperties || []),
+				],
+				neo4jResult: result,
+			},
+		]);
```

Comment:
Could we make these events via some event creation function which is exported by events.js with required arguments e.g action, code and type, and other optional object argument?  This is because making event parameters has pretty much kinds, Is it worth to do?

Reply:
It might be. More or less that is already what's happening internally - `broadcast()` calls the `makeEvents()` function. But for all use cases apart from absorb there is just a single call, so we would end up replacing `broadcast(...)` with `broadcast(makeEvents(...))` in most handlers.

Reply:
Ah! I agree with you.